imxrt1060-hal/src/gpio.rs

```diff
@@ -212,6 +219,84 @@ where
     }
 }
 
+/// GPIO input interrupt configurations.
+///
+/// These configurations do not take effect until
+/// GPIO input interrupts are enabled.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[repr(u32)]
+pub enum InterruptConfiguration {
+    // TODO isn't it a bit repetitive to have "Sensitive" on all here?
+    LowLevelSensitive = 0,
+    HighLevelSensitive = 1,
+    RisingEdgeSensitive = 2,
+    FallingEdgeSensitive = 3,
+    EitherEdgeSensitive = 4,
+}
+
+impl<P> GPIO<P, Input>
+where
+    P: Pin,
+{
+    /// Enable (`true`) or disable (`false`) interrupts for this GPIO input.
+    pub fn set_interrupt_enable(&mut self, enable: bool) {
+        cortex_m::interrupt::free(|_| unsafe {
+            ral::modify_reg!(ral::gpio, self.register_block(), IMR, |imr| if enable {
+                imr | self.mask()
+            } else {
+                imr & !self.mask()
+            })
+        });
+    }
+
+    /// Indicates if interrupts are (`true`) or are not (`false`) enabled for this GPIO input.
+    pub fn is_interrupt_enabled(&self) -> bool {
+        unsafe { ral::read_reg!(ral::gpio, self.register_block(), IMR) & self.mask() != 0u32 }
+    }
+
+    /// Set the interrupt configuration for this GPIO input.
+    pub fn set_interrupt_configuration(&mut self, interrupt_configuration: InterruptConfiguration) {
+        if InterruptConfiguration::EitherEdgeSensitive == interrupt_configuration {
+            unsafe {
+                ral::modify_reg!(ral::gpio, self.register_block(), EDGE_SEL, |edge_sel| {
+                    edge_sel | self.mask()
+                });
+            }
+        } else {
+            unsafe {
+                ral::modify_reg!(ral::gpio, self.register_block(), EDGE_SEL, |edge_sel| {
+                    edge_sel & !self.mask()
+                });
+            }
+            // TODO is it ok to assume the numeric value of the enum is the icr?
```

Comment:
In your async code you had an explicit mapping from the enum to the `icr`. Should we keep that?

Reply:
I like your `as u32` approach! Wasn't relying on a `#[repr(u32)]` enum in the async code, though an `as` cast might have yielded the same values.